From: Ralf Baechle <ralf@linux-mips.org>
Date: Mon, 16 Oct 2006 15:49:37 +0000 (+0100)
Subject: [MIPS] Use compat_sys_mount.
X-Git-Tag: linux-2.6.18.1^0~1
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=6b80e7725f91669a39853b5b4cd9c6599acce7ef;hp=55ceaed731ab2d60d855ebec59586fa3a327eebb

[MIPS] Use compat_sys_mount.

This fixes mount problems with smbfs, ncpfs and NFSv4.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from 0b2177d236a233461f9830b51d8fc4f71ad60d43 commit)
---

diff --git a/arch/mips/kernel/scall64-n32.S b/arch/mips/kernel/scall64-n32.S
index f25c2a2..7960f97 100644
--- a/arch/mips/kernel/scall64-n32.S
+++ b/arch/mips/kernel/scall64-n32.S
@@ -280,7 +280,7 @@ EXPORT(sysn32_call_table)
 	PTR	sys_sync
 	PTR	sys_acct
 	PTR	sys32_settimeofday
-	PTR	sys_mount			/* 6160 */
+	PTR	compat_sys_mount		/* 6160 */
 	PTR	sys_umount
 	PTR	sys_swapon
 	PTR	sys_swapoff
diff --git a/arch/mips/kernel/scall64-o32.S b/arch/mips/kernel/scall64-o32.S
index 288ee4a..51284b8 100644
--- a/arch/mips/kernel/scall64-o32.S
+++ b/arch/mips/kernel/scall64-o32.S
@@ -226,7 +226,7 @@ sys_call_table:
 	PTR	sys_ni_syscall			/* was sys_stat */
 	PTR	sys_lseek
 	PTR	sys_getpid			/* 4020 */
-	PTR	sys_mount
+	PTR	compat_sys_mount
 	PTR	sys_oldumount
 	PTR	sys_setuid
 	PTR	sys_getuid
